### DISCUSSION TITLE
Vineyard 是否支持java 语言

### DISCUSSION THREAD

**Discussion by yg000** [v6d-io/v6d#1070] [General]
您好。想请教一下，Vineyard 是否支持java 语言？谢谢。

**Comment by dashanji** [comment]
您好，目前java sdk还不太完善，您可以尝试python或者C++ SDK，具体可参考官网文档 https://v6d.io/notes/getting-started.html

**Comment by yg000** [comment]
您好，收到您的回复，十分感谢！




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "v6d-io/v6d"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年11月23日(星期三) 下午3:46
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [v6d-io/v6d] Vineyard 是否支持java 语言 (Discussion #1070)





 
您好，目前java sdk还不太完善，您可以尝试python或者C++ SDK，具体可参考官网文档 https://v6d.io/notes/getting-started.html

**Comment by yg000** [comment]
您好，我还有一个问题，Vineyard&nbsp; 支持跨语言 内存数据调用吗? (例如python 和 C++ 之间)，感谢！




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "v6d-io/v6d"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年11月23日(星期三) 下午3:46
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [v6d-io/v6d] Vineyard 是否支持java 语言 (Discussion #1070)





 
您好，目前java sdk还不太完善，您可以尝试python或者C++ SDK，具体可参考官网文档 https://v6d.io/notes/getting-started.html

**Reply by sighingnow** [reply]
> 您好，我还有一个问题，Vineyard&nbsp; 支持跨语言 内存数据调用吗? (例如python 和 C++ 之间)，感谢！

有的。从C++build进vineyard的data structure，可以从python client去get，tutorial里有一些简单的例子，文档正在丰富中。

**Comment by sighingnow** [comment]
> 您好。想请教一下，Vineyard 是否支持java 语言？谢谢。

Java有初步的支持，

- 有java client的实现
- 有和apache-arrow java sdk的integration
- 有初步的spark的integration

**Reply by MikuSugar** [reply]
@sighingnow 你好，请问现在对Java的支持如何，没有在官网找到相关文档。谢谢🙏

**Reply by sighingnow** [reply]
您好，近期会整理javadoc，目前的实现包括 java client 和 arrow data structures的支持，可以参考：

- https://github.com/v6d-io/v6d/tree/main/java/modules
- https://github.com/v6d-io/v6d/tree/main/java/core
